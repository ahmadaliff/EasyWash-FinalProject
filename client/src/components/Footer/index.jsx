import { FormattedMessage } from 'react-intl';

import { Copyright, Email, GitHub, LinkedIn } from '@mui/icons-material';

import classes from '@components/Footer/style.module.scss';

const Footer = () => (
  <footer className={classes.footer} data-testid="footer">
    <hr className={classes.line} />
    <div className={classes.footerContent}>
      <div className={classes.leftContent}>
        <img src="/longLogo.svg" alt="logo" className={classes.logo} />
        <span className={classes.develop}>
          <FormattedMessage id="app_develop_by" /> :
        </span>
        <span className={classes.link} data-testid="linkedIn">
          <LinkedIn />
          <a
            className={classes.anchorLink}
            target="_blank"
            href="https://www.linkedin.com/in/alifahmaaad/"
            rel="noreferrer"
          >
            Ahmad Alif Sofian
          </a>
        </span>
        <span className={classes.link} data-testid="github">
          <GitHub />
          <a href="https://github.com/alifahmaaad" target="_blank" className={classes.anchorLink} rel="noreferrer">
            Ahmad Alif Sofian
          </a>
        </span>
        <span className={classes.link} data-testid="email">
          <Email />
          <a className={classes.anchorLink} href="mailto:alif12sofian@gmail.com">
            alif12sofian@gmail.com
          </a>
        </span>
      </div>

      <div className={classes.rightContent}>
        <img className={classes.ilustrator} src="/footer.svg" alt="https://storyset.com/" />
        <div className={classes.bootomText}>
          <Copyright />
          2023, EasyWash - Ahmad Alif Sofian
        </div>
      </div>
    </div>
  </footer>
);

export default Footer;
