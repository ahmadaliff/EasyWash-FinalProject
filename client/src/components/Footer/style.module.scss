@import '@styles/base';

.footer {
  @include directionY;
  align-items: center;
  justify-content: center;
  padding: 0.75rem;
  width: 100%;
  .line {
    width: 100%;
  }
  .footerContent {
    @include directionY;
    @include apply-at-md-size {
      @include directionX;
    }
    padding: 1rem;
    align-items: center;
    justify-content: space-around;
    width: 100%;
    gap: 2rem;
    .leftContent {
      @include directionY;
      gap: 0.5rem;
      align-self: flex-start;
      @include apply-at-md-size {
        align-self: auto;
      }
      .logo {
        position: relative;
        left: -0.75rem;
      }
      .develop {
        font-size: large;
      }
      .link {
        @include directionX;
        align-items: center;
        gap: 0.5rem;
        .anchorLink {
          font-size: small;
          color: var(--color-text);
        }
      }
    }
    .rightContent {
      @include directionY;
      align-items: center;
      gap: 1rem;
      max-width: 40rem;
      .ilustrator {
        width: 15rem;
        @include apply-at-md-size {
          width: 25rem;
        }
      }
    }
  }
  .bootomText {
    @include directionX;
    align-items: center;
    font-size: small;
    color: var(--color-primary);
    svg {
      width: 1rem;
      height: 1rem;
    }
  }
}
