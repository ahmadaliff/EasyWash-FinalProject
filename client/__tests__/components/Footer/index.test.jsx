import Footer from '@components/Footer';
import { render, screen } from '@utils/testHelper';

let footer;

beforeEach(() => {
  footer = render(<Footer />);
});

describe('Footer Component', () => {
  test('Correct render', () => {
    const { getByTestId } = footer;
    expect(getByTestId('footer')).toBeInTheDocument();
    expect(getByTestId('linkedIn')).toBeInTheDocument();
    expect(getByTestId('github')).toBeInTheDocument();
    expect(getByTestId('email')).toBeInTheDocument();
  });
  test('Should be render text', () => {
    expect(screen.getByText('alif12sofian@gmail.com')).toBeInTheDocument();
    expect(screen.getByText('2023, EasyWash - Ahmad Alif Sofian')).toBeInTheDocument();
  });
  test('Should match with snapshot', () => {
    expect(footer).toMatchSnapshot();
  });
});
