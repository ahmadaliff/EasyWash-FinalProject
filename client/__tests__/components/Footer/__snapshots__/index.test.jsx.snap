// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer"
        data-testid="footer"
      >
        <hr
          class="line"
        />
        <div
          class="footerContent"
        >
          <div
            class="leftContent"
          >
            <img
              alt="logo"
              class="logo"
              src="/longLogo.svg"
            />
            <span
              class="develop"
            >
              Dikembangkan Oleh
               :
            </span>
            <span
              class="link"
              data-testid="linkedIn"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LinkedInIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                />
              </svg>
              <a
                class="anchorLink"
                href="https://www.linkedin.com/in/alifahmaaad/"
                rel="noreferrer"
                target="_blank"
              >
                Ahmad Alif Sofian
              </a>
            </span>
            <span
              class="link"
              data-testid="github"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="GitHubIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                />
              </svg>
              <a
                class="anchorLink"
                href="https://github.com/alifahmaaad"
                rel="noreferrer"
                target="_blank"
              >
                Ahmad Alif Sofian
              </a>
            </span>
            <span
              class="link"
              data-testid="email"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EmailIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
              <a
                class="anchorLink"
                href="mailto:alif12sofian@gmail.com"
              >
                alif12sofian@gmail.com
              </a>
            </span>
          </div>
          <div
            class="rightContent"
          >
            <img
              alt="https://storyset.com/"
              class="ilustrator"
              src="/footer.svg"
            />
            <div
              class="bootomText"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CopyrightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.88 9.14c1.28.06 1.61 1.15 1.63 1.66h1.79c-.08-1.98-1.49-3.19-3.45-3.19C9.64 7.61 8 9 8 12.14c0 1.94.93 4.24 3.84 4.24 2.22 0 3.41-1.65 3.44-2.95h-1.79c-.03.59-.45 1.38-1.63 1.44-1.31-.04-1.86-1.06-1.86-2.73 0-2.89 1.28-2.98 1.88-3zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
              2023, EasyWash - Ahmad Alif Sofian
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer"
      data-testid="footer"
    >
      <hr
        class="line"
      />
      <div
        class="footerContent"
      >
        <div
          class="leftContent"
        >
          <img
            alt="logo"
            class="logo"
            src="/longLogo.svg"
          />
          <span
            class="develop"
          >
            Dikembangkan Oleh
             :
          </span>
          <span
            class="link"
            data-testid="linkedIn"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LinkedInIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
              />
            </svg>
            <a
              class="anchorLink"
              href="https://www.linkedin.com/in/alifahmaaad/"
              rel="noreferrer"
              target="_blank"
            >
              Ahmad Alif Sofian
            </a>
          </span>
          <span
            class="link"
            data-testid="github"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="GitHubIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
              />
            </svg>
            <a
              class="anchorLink"
              href="https://github.com/alifahmaaad"
              rel="noreferrer"
              target="_blank"
            >
              Ahmad Alif Sofian
            </a>
          </span>
          <span
            class="link"
            data-testid="email"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EmailIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
            <a
              class="anchorLink"
              href="mailto:alif12sofian@gmail.com"
            >
              alif12sofian@gmail.com
            </a>
          </span>
        </div>
        <div
          class="rightContent"
        >
          <img
            alt="https://storyset.com/"
            class="ilustrator"
            src="/footer.svg"
          />
          <div
            class="bootomText"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CopyrightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.88 9.14c1.28.06 1.61 1.15 1.63 1.66h1.79c-.08-1.98-1.49-3.19-3.45-3.19C9.64 7.61 8 9 8 12.14c0 1.94.93 4.24 3.84 4.24 2.22 0 3.41-1.65 3.44-2.95h-1.79c-.03.59-.45 1.38-1.63 1.44-1.31-.04-1.86-1.06-1.86-2.73 0-2.89 1.28-2.98 1.88-3zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            2023, EasyWash - Ahmad Alif Sofian
          </div>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
